soroban-env-host/src/native_contract/token/balance.rs

```diff
@@ -107,16 +106,16 @@ pub fn spend_balance_no_freeze_check(
 
 // Metering: covered by components.
 pub fn spend_balance(e: &Host, id: Identifier, amount: i128) -> Result<(), HostError> {
-    let is_frozen = read_state(e, id.clone())?;
+    let is_frozen = is_deauthorized(e, id.clone())?;
     if is_frozen {
         return Err(e.err_status_msg(ContractError::BalanceFrozenError, "balance is frozen"));
     }
 
-    spend_balance_no_freeze_check(e, id, amount)
+    spend_balance_no_authorization_check(e, id, amount)
 }
 
 // Metering: *mostly* covered by components. Not sure about `try_into_val`.
-pub fn read_state(e: &Host, id: Identifier) -> Result<bool, HostError> {
+pub fn is_deauthorized(e: &Host, id: Identifier) -> Result<bool, HostError> {
```

Comment:
I think positive bool fn/variable names are more readable (especially when e.g. negated). So I think `is_authorized` would be better.

Reply:
Done. Also updated is_frozen to authorized.